http_handlers.go

```diff
@@ -102,6 +141,109 @@ func getMetadata(filter string) http.HandlerFunc {
 	}
 }
 
+func ec2Handler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	logger.Debug("Calling ec2Handler ")
+	userIP := getIPFromRequest(r)
+	if userIP == "" {
+		return
+	}
+
+	metrics.MetadataRequests.Inc()
+	logger.With("userIP", userIP).Info("Actual IP is: ")
+	hw, err := getByIP(context.Background(), hegelServer, userIP) // returns hardware data as []byte
+	if err != nil {
+		metrics.Errors.WithLabelValues("metadata", "lookup").Inc()
+		logger.Info("Error in finding hardware: ", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	res, err := processEC2Query(r.URL.Path)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("404 not found"))
```

Comment:
This is actually a different error code + message compared to Kant, since the original was really vague: `{"status":500,"message":"Something went wrong..."}`. 
Do you have a suggestion for this?